### PR TITLE
Bump Terraform provider version to v4.56.0

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Depends on: https://github.com/upbound/upjet/pull/172

This PR bumps the underlying Terraform provider version to `v4.56.0`.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested resources reported by `crddiff`:
- `Cluster.containerazure`: https://github.com/upbound/provider-gcp/actions/runs/4378403601
- `sql.DatabaseInstance `: https://github.com/upbound/provider-gcp/actions/runs/4378404381

### CODE REVIEW

**Comment by ulucinar** [comment]
/test-examples="examples/cloudplatform/serviceaccount.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/containerazure/cluster.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/sql/instance.yaml"

**Comment by ulucinar** [comment]
Two MR definitions are reported to have breaking changes: `containerazure.gcp.upbound.io_clusters` and `sql.gcp.upbound.io_databaseinstances`.

No resources have been reported by the schemadiff tool.

**Comment by ulucinar** [comment]
/test-examples="examples/containerazure/cluster.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/sql/instance.yaml"

**Review by sergenyalcin — APPROVED** [review]
Thanks @ulucinar LGTM!

File: config/sql/config.go [hunk 1]
```diff
@@ -31,6 +31,9 @@ const (
 // ResourceConfigurators.
 func Configure(p *config.Provider) { //nolint:gocyclo
 	p.AddResourceConfigurator("google_sql_database_instance", func(r *config.Resource) {
+		// we need to workaround the newly added (with v4.56.0)
+		// of `Optional=True` in the native provider
+		config.MoveToStatus(r.TerraformResource, "instance_type")
```
Comment (ulucinar) [inline config/sql/config.go:36]: In `v4.56.0` of the Terraform Google provider, `Optional: true` has been added to the native schema of the `instance_type` attribute, which moves the generated field to `spec`. We need to move it back to `status`.

**Pull request merged**